app/models/mixins/live_metrics_mixin.rb

```diff
@@ -0,0 +1,39 @@
+module LiveMetricsMixin
+  extend ActiveSupport::Concern
+
+  class MetricValidationError < RuntimeError; end
+
+  included do
+    def metrics_available
+      raise 'LiveMetricsMixing: metrics_available missing'
+    end
+
+    def metric(_metric, _start_time, _end_time, _interval)
+      raise 'LiveMetricsMixing: metric method missing'
+    end
+
+    def metrics(*metrics, start_time, end_time, interval)
+      metrics = metrics_available if metrics.empty?
+      processed = Hash.new { |h, k| h[k] = {} }
+      metrics.each do |m|
+        values = metric(m, start_time, end_time, interval)
+        values.each do |k, a|
+          processed[k].merge!(a)
```

Comment:
True, I think I could add a block in the merge call and simplify the complexity.
Another Java practice brought to Ruby language, my bad.